Compiler error on compiling nested fixed-size array

### Describe the bug

Nesting a fixed-size array inside an array , then re-assigning the first element cannot be compiled.

### Reproduction Steps

```v
module main
fn main() {
	mut net := [][2]int{len: 100}
	net[0] = [1, 1]!
}
```
Compiling this (v -cg file.v) produces a compiler error:

C:/.../Temp/v_0/ok.5538097916378980171.tmp.c:493: warning: WINVER redefined
C:/.../Temp/v_0/ok.5538097916378980171.tmp.c:6860: warning: implicit declaration of function 'tcc_backtrace'
C:/.../Temp/v_0/ok.5538097916378980171.tmp.c:12831: error: ',' expected (got ")")


### Expected Behavior

To assign the fixed-size array [1, 1]! to element 0 of the array "net".

### Current Behavior

Compiler error.

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.3 08189d6

### Environment details (OS name and version, etc.)

V full version: V 0.4.3 08189d6
OS: windows, Microsoft Windows 10 Pro v19045 64-Bit                    
Processor: 32 cpus, 64bit, little endian,                              
                                                                        
thirdparty/tcc status: thirdparty-windows-amd64 a39eb79b  
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -429,8 +429,10 @@ fn (mut g Gen) assign_stmt(node_ ast.AssignStmt) {
 				pos := g.out.len
 				g.expr(left)
 
-				if g.is_arraymap_set && g.arraymap_set_pos > 0 {
-					g.go_back_to(g.arraymap_set_pos)
+				if g.is_arraymap_set && g.arraymap_set_pos >= 0 {
+					if g.arraymap_set_pos > 0 {
+						g.go_back_to(g.arraymap_set_pos)
+					}
 					g.write(', &${v_var})')
 					g.is_arraymap_set = false
 					g.arraymap_set_pos = 0

--- a/vlib/v/gen/c/index.v
+++ b/vlib/v/gen/c/index.v
@@ -224,10 +224,12 @@ fn (mut g Gen) index_of_array(node ast.IndexExpr, sym ast.TypeSymbol) {
 								else {}
 							}
 				*/
-				if need_wrapper {
-					g.write(', &(${elem_type_str}[]) { ')
-				} else {
-					g.write(', &')
+				if elem_sym.kind != .array_fixed {
+					if need_wrapper {
+						g.write(', &(${elem_type_str}[]) { ')
+					} else {
+						g.write(', &')
+					}
 				}
 			} else {
 				// `x[0] *= y`

--- a/vlib/v/tests/array_with_fixed_array_test.v
+++ b/vlib/v/tests/array_with_fixed_array_test.v
@@ -1,0 +1,16 @@
+fn fixed() [2]int {
+	return [9, 9]!
+}
+
+fn test_main() {
+	arr := [1, 2]!
+	mut net := [][2]int{len: 4}
+	net[1] = [1, 1]!
+	net[0] = net[1]
+	net[2] = arr
+	net[3] = fixed()
+	assert net[0] == [1, 1]!
+	assert net[1] == [1, 1]!
+	assert net[2] == [1, 2]!
+	assert net[3] == [9, 9]!
+}

